Improve ElectronIDExternalProducer

Dr15Jones opened the pull request:
#### PR description:

- made it a global module
- updated helpers to have const member functions
- reduced memory churn by setting values from ParameterSet during
  construction
- fixed a 10 year old bug for case where CutBasedElectronID is supposed to retrieve the BeamSpot

#### PR validation:

Code compiles. Running a trivial RECO workflow did succeed.

NOTE: this should be a purely technical change and is not expected to change results for the cases we use.

This WILL change behavior IF an old version is used, but it returns the code to the behavior it had before the bug was introduced at the time this code was updated to call `consumes`.

fixes #46866

Dr15Jones committed:
Improved ElectronIDExternalProducer

- made it a global module
- updated helpers to have const member functions
- reduced memory churn by setting values from ParameterSet during
  construction

Dr15Jones committed:
Fix retrieval of beam spot in CutBasedElectronID

Properly do consumes for beam spot if needed.
Switched to Event::get methods which will properly type check.

Dr15Jones commented:
please test

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"2f9814e68c22d978574648325f176542b6f8362f":{"files":["RecoEgamma/ElectronIdentification/plugins/CutBasedElectronID.cc","RecoEgamma/ElectronIdentification/plugins/CutBasedElectronID.h"],"squashed":false,"time":1733326911},"5607b8a27f29fb7ff6e5c3d3bf5717a3fa39a0b4":{"files":["RecoEgamma/ElectronIdentification/plugins/ClassBasedElectronID.cc","RecoEgamma/ElectronIdentification/plugins/ClassBasedElectronID.h","RecoEgamma/ElectronIdentification/plugins/CutBasedElectronID.cc","RecoEgamma/ElectronIdentification/plugins/CutBasedElectronID.h","RecoEgamma/ElectronIdentification/plugins/ElectronIDAlgo.h","RecoEgamma/ElectronIdentification/plugins/ElectronIDExternalProducer.h","RecoEgamma/ElectronIdentification/plugins/ElectronIDSelector.h","RecoEgamma/ElectronIdentification/plugins/ElectronIDSelectorCutBased.cc","RecoEgamma/ElectronIdentification/plugins/ElectronIDSelectorCutBased.h","RecoEgamma/ElectronIdentification/plugins/PTDRElectronID.cc","RecoEgamma/ElectronIdentification/plugins/PTDRElectronID.h"],"squashed":false,"time":1733320922}},"emoji":{},"last_seen_sha":"2f9814e68c22d978574648325f176542b6f8362f","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46868/42897

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones for master.

It involves the following packages:

- RecoEgamma/ElectronIdentification (**reconstruction**)


@jfernan2, @mandrenguyen can you please review it and eventually sign? Thanks.
@Prasant1993, @Sam-Harper, @a-kapoor, @afiqaize, @jainshilpi, @lgray, @missirol, @ram1123, @sameasy, @sobhatta, @valsdav, @varuns23 this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones commented:
There is an additional change I'd like to do. I would like to change  `CutBasedElectronID` so it uses the _strategy pattern_ for the versioning. That is, distinct value of `version_` would be handled via polymorphism instead of large number of `if` statements. This would allow the parameters to be extracted at construction time instead of being called for each electron individually (which is time and memory intensive).

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43245/summary.html
**COMMIT**: 2f9814e68c22d978574648325f176542b6f8362f
**CMSSW**: CMSSW_15_0_X_2024-12-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46868/43245/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

- 2024.303001```DAS Error```
- 2024.300001```DAS Error```
- 2024.203001```DAS Error```
<details>
<summary>Expand to see more relval errors ...</summary>

- 2024.103001
- 2024.302001
- 2024.101001
- 2024.102001
- 2024.100001
</details>



## Comparison Summary



There are some workflows for which there are errors in the baseline:
 2024.202001 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 39 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-1100+896277/65969/validateJR.html)
 - DQMHistoTests: Total files compared: 45
 - DQMHistoTests: Total histograms compared: 3393039
 - DQMHistoTests: [Total failures: 1873](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-1100+896277/65969/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3391146
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 44 files compared)
 - Checked 198 log files, 168 edm output root files, 45 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-1100+896277/65969/triggerResults): no differences found

jfernan2 commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43269/summary.html
**COMMIT**: 2f9814e68c22d978574648325f176542b6f8362f
**CMSSW**: CMSSW_15_0_X_2024-12-04-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46868/43269/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



There are some workflows for which there are errors in the baseline:
 2022.000001 step 1
2022.002001 step 1
2023.002001 step 1
2024.000001 step 1
2024.101001 step 1
2024.202001 step 1
2024.303001 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 32 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-2300+896277/65987/validateJR.html)
 - DQMHistoTests: Total files compared: 39
 - DQMHistoTests: Total histograms compared: 2957317
 - DQMHistoTests: [Total failures: 1526](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-2300+896277/65987/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2955771
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 38 files compared)
 - Checked 174 log files, 144 edm output root files, 39 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-04-2300+896277/65987/triggerResults): found differences in 1 / 37 workflows

jfernan2 commented:
> There is an additional change I'd like to do. I would like to change `CutBasedElectronID` so it uses the _strategy pattern_ for the versioning. That is, distinct value of `version_` would be handled via polymorphism instead of large number of `if` statements. This would allow the parameters to be extracted at construction time instead of being called for each electron individually (which is time and memory intensive).

@Dr15Jones are you going to include that change in this PR? Thanks

Dr15Jones commented:
> are you going to include that change in this PR? Thanks

I'm working on the changes but they are extensive. I'll probably have the code done today but I'm still trying to figure out a good way to test.

Sam-Harper commented:
Before too much effort is expended on this. My understanding is that this module died in Run 1.  Is this being used somewhere?  E/gamma has used VID since Run2 and this is deprecated to my understanding. Is there something I miss? Can we not just delete it?

Dr15Jones commented:
I discovered this module by running a Run3 workflow. The configuration header had
```
# Source: /local/reps/CMSSW/CMSSW/Configuration/Applications/python/ConfigBuilder.py,v 
# with command line options: step3 -s RAW2DIGI,L1Reco,RECO,RECOSIM,SKIM:LogError+LogErrorMonitor,PAT,NANO,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@nanoAODDQM --conditions auto:phase1_2024_realistic --datatier RECO,MINIAODSIM,NANOAODSIM,DQMIO -n 10 --eventcontent RECOSIM,MINIAODSIM,NANOEDMAODSIM,DQM --geometry DB:Extended --era Run3_2024 --customise Validation/Performance/TimeMemorySummary.customiseWithTimeMemorySummary --filein file:step2.root --fileout file:step3.root
```

jfernan2 commented:
FYI: @SanghyunKo @cochando

Sam-Harper commented:
In my opinion, the action should be to delete this module and remove it from where ever its being used. The IDs are at least 12 years out of date. 

@cms-sw/egamma-pog-l2

Dr15Jones commented:
The labels for the module in the workflow are : `eidLoose` , `JPTeidTight` and `eidTight`.

Sam-Harper commented:
Thanks Chris

So on this, I think we need to understand what JetMET are using this ancient ID for and whether they could move to another ID

@cms-sw/jetmet-pog-l2

Dr15Jones committed:
Modified CutBasedElectronID to use the strategy pattern

Each version is now its own class.
Added unit test to show the producer has no gross failures.

Dr15Jones committed:
CutBasedElectronID now gets parameters at construction time

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46868/42933

cmsbuild commented:
Pull request #46868 was updated. @cmsbuild, @jfernan2, @mandrenguyen can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones commented:
I found another bug related to ElectronIDExternalProducer. The configuration file generated by cmsDriver.py has the wrong parameters inside the PSet labeled `classbasedtightEleIDCutsV02`. The actual V02 algorithm fails because it can't find the parameters it needs.

cmsbuild commented:
-1

**Failed Tests**:   ClangBuild 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43304/summary.html
**COMMIT**: c0f780e0a92e2f6b344a8acb8e36a48501b1fb61
**CMSSW**: CMSSW_15_0_X_2024-12-06-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46868/43304/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

Dr15Jones committed:
Fix clang compilation issue

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46868/42934

cmsbuild commented:
Pull request #46868 was updated. @cmsbuild, @jfernan2, @mandrenguyen can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals RelVals-INPUT AddOn 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/summary.html
**COMMIT**: a8f327f5337e3ba0524e2367f9a368b60ccc3b37
**CMSSW**: CMSSW_15_0_X_2024-12-06-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46868/43305/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 4 errors in the following unit tests:

<pre>---> test test_MC_22_crosscheck had ERRORS
---> test test_MC_23_crosscheck had ERRORS
---> test test_MC_23_setup had ERRORS
and more ...
</pre>

## RelVals

- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/135.4_ZEEFS_13)
```
----- Begin Fatal Exception 06-Dec-2024 22:48:41 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/1306.0_SingleMuPt1_UP15)
```
----- Begin Fatal Exception 06-Dec-2024 22:50:27 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
- [2024.202001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2024.202001_RunJetMET02024D_10k)
```
----- Begin Fatal Exception 06-Dec-2024 22:53:36 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/24834.911_TTbar_14TeV+Run4D98_DD4hep)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/4.53_RunPhoton2012B)
- [29634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/29634.0_TTbar_14TeV+Run4D110)
- [2024.303001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2024.303001_RunDisplacedJet2024E_10k)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/5.1_TTbarFS)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [2024.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2024.000001_RunZeroBias2024B_10k)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/13234.0_TTbar_14TeV+2022FS)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/1000.0_RunMinBias2011A)
- [2024.101001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2024.101001_RunBTagMu2024C_10k)
- [2022.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2022.000001_RunJetHT2022C_10k)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/11634.0_TTbar_14TeV+2022)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/136.874_RunEGamma2018C)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [16834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/16834.0_TTbar_14TeV+2025)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [29700.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/29700.0_CloseByPGun_CE_H_Coarse_Scint+Run4D110)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/136.731_RunSinglePh2016B)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/12846.0_ZEE_14+2024)
- [29696.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/29696.0_CloseByPGun_CE_E_Front_120um+Run4D110)
- [2023.002001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2023.002001_RunZeroBias2023D_10k)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/25202.0_TTbar_13)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/25.0_TTbar)
- [2022.002001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/2022.002001_RunZeroBias2022C_10k)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/1330.0_ZMM_13)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/139.001_RunMinimumBias2021)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/140.56_RunHI2018)
- [29834.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/29834.999_TTbar_14TeV+Run4D110PU_PMXS1S2PR)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/1001.0_RunMinBias2011A)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [29634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/29634.911_TTbar_14TeV+Run4D110_DD4hep)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
</details>



## RelVals-INPUT

- [138.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/138.5_ExpressCollisions2021)```138.5_ExpressCollisions2021/step2_ExpressCollisions2021.log```
- [138.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/138.4_PromptCollisions2021)```138.4_PromptCollisions2021/step2_PromptCollisions2021.log```
- [142.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/142.903_RunUPC2016PA)```142.903_RunUPC2016PA/step2_RunUPC2016PA.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [142.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/142.902_RunUPC2023)
- [142.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/142.901_RunUPC2023)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/134.0_MinimumBiasBS)
- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/4.6_MinimumBias2010A)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1000.0_RunMinBias2011A)
- [141.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/141.902_RunUPC2023)
- [1001.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1001.4_RunDoubleMuon2022B)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/144.6_MinimumBias2010A)
- [140.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/140.61_RunHI2022)
- [2500.245](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/2500.245_hcalDPGCalibNANO140Xrun3)
- [2500.244](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/2500.244_hcalDPGNANO140Xrun3)
- [136.902](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/136.902_RunDoubleMuon2016H)
- [136.903](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/136.903_RunDoubleMuon2017B)
- [136.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/136.901_RunDoubleMuon2016C)
- [1003.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1003.0_RunMinBias2012A)
- [136.904](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/136.904_RunDoubleMuon2018C)
- [1040.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1040.0_RunZeroBias2017F)
- [141.901](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/141.901_RunUPC2023)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/140.56_RunHI2018)
- [140.57](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/140.57_RunHI2018Reduced)
- [1001.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1001.2_RunZeroBias2017F)
- [1001.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1001.3_RunSingleMuon2022B)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1001.0_RunMinBias2011A)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1002.0_RRD)
- [1002.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1002.3_RunZeroBias2022B)
- [1002.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/runTheMatrixINPUT-results/1002.4_RunDoubleMuon2022B)
</details>



## AddOn Tests

- [fastsim](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/fastsim)
```
----- Begin Fatal Exception 06-Dec-2024 22:48:16 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
- [fastsim1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/fastsim1)
```
----- Begin Fatal Exception 06-Dec-2024 22:48:20 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
- [fastsim2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/fastsim2)
```
----- Begin Fatal Exception 06-Dec-2024 22:48:23 CET-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=EleIdCutBasedExtProducer label='eidRobustHighEnergy'
Exception Message:
MissingParameter: Parameter 'cutiso_sum' not found.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_mc_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_Fake)
- [hlt_mc_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_Fake1)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_Fake2)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_GRun)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_HIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_PIon)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_PRef)
- [hlt_mc_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_mc_Special)
- [hlt_data_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_Fake)
- [hlt_data_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_Fake1)
- [hlt_data_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_Fake2)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_GRun)
- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_HIon)
- [hlt_data_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_PIon)
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_PRef)
- [hlt_data_Special](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43305/addOnTests/hlt_data_Special)
</details>

Dr15Jones committed:
Better handling of robust case for CutBasedElectronID

Broke robust and classbased into their own strategies this also
makes sure parameters are only read if needed.

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46868/42940

cmsbuild commented:
Pull request #46868 was updated. @jfernan2, @mandrenguyen can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT AddOn 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/summary.html
**COMMIT**: 342e295d086419a41447d0f3328943eb4db1559b
**CMSSW**: CMSSW_15_0_X_2024-12-07-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46868/43309/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/135.4_ZEEFS_13)```135.4_ZEEFS_13/step1_ZEEFS_13.log```
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/9.0_Higgs200ChargedTaus)```9.0_Higgs200ChargedTaus/step3_Higgs200ChargedTaus.log```
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/1306.0_SingleMuPt1_UP15)```1306.0_SingleMuPt1_UP15/step3_SingleMuPt1_UP15.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/7.3_CosmicsSPLoose2018)
- [29700.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/29700.0_CloseByPGun_CE_H_Coarse_Scint+Run4D110)
- [2024.202001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2024.202001_RunJetMET02024D_10k)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/24834.911_TTbar_14TeV+Run4D98_DD4hep)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/4.53_RunPhoton2012B)
- [29634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/29634.0_TTbar_14TeV+Run4D110)
- [2024.303001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2024.303001_RunDisplacedJet2024E_10k)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/5.1_TTbarFS)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [2024.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2024.000001_RunZeroBias2024B_10k)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/13234.0_TTbar_14TeV+2022FS)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/1000.0_RunMinBias2011A)
- [2024.101001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2024.101001_RunBTagMu2024C_10k)
- [2022.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2022.000001_RunJetHT2022C_10k)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/136.874_RunEGamma2018C)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/11634.0_TTbar_14TeV+2022)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [16834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/16834.0_TTbar_14TeV+2025)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [8.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/8.0_BeamHalo)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/136.731_RunSinglePh2016B)
- [2500.201](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2500.201_NANOmc140X)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/12846.0_ZEE_14+2024)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/4.22_RunCosmics2011A)
- [29696.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/29696.0_CloseByPGun_CE_E_Front_120um+Run4D110)
- [2023.002001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2023.002001_RunZeroBias2023D_10k)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/25202.0_TTbar_13)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/25.0_TTbar)
- [2022.002001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/2022.002001_RunZeroBias2022C_10k)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/1330.0_ZMM_13)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/139.001_RunMinimumBias2021)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/140.56_RunHI2018)
- [29834.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/29834.999_TTbar_14TeV+Run4D110PU_PMXS1S2PR)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [29634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/29634.911_TTbar_14TeV+Run4D110_DD4hep)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
</details>



## RelVals-INPUT

- [159.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrixINPUT-results/159.01_HydjetQ_reminiaodPbPb2022_INPUT)```159.01_HydjetQ_reminiaodPbPb2022_INPUT/step2_HydjetQ_reminiaodPbPb2022_INPUT.log```
- [2500.024](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrixINPUT-results/2500.024_NANOmcUL18reMINI)```2500.024_NANOmcUL18reMINI/step2_NANOmcUL18reMINI.log```
- [2500.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrixINPUT-results/2500.023_NANOmcUL17reMINI)```2500.023_NANOmcUL17reMINI/step2_NANOmcUL17reMINI.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [2500.022](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrixINPUT-results/2500.022_NANOmcUL16reMINI)
- [2500.021](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/runTheMatrixINPUT-results/2500.021_NANOmcUL16APVreMINI)
- 2024.003001
- 2024.100001
</details>



## AddOn Tests

- [fastsim](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/addOnTests/fastsim)
```
UNKNOWN
```
- [fastsim1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/addOnTests/fastsim1)
```
UNKNOWN
```
- [fastsim2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43309/addOnTests/fastsim2)
```
UNKNOWN
```

Dr15Jones commented:
The jobs are dying in python saying `Validation.HGCalValidation.hgcalSimHitValidationEE_cfi` is missing.

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-896277/43318/summary.html
**COMMIT**: 342e295d086419a41447d0f3328943eb4db1559b
**CMSSW**: CMSSW_15_0_X_2024-12-08-0000/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46868/43318/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-08-0000+896277/66023/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-08-0000+896277/66023/validateJR.html)
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3510017
 - DQMHistoTests: [Total failures: 417](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-08-0000+896277/66023/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3509580
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 45 files compared)
 - Checked 202 log files, 172 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_0_X_2024-12-08-0000+896277/66023/triggerResults): no differences found

jfernan2 commented:
@lnurfikri89 as JetMET RECO contact, please comment on;

> 
> So on this, I think we need to understand what JetMET are using this ancient ID for and whether they could move to another ID
> 
> @cms-sw/jetmet-pog-l2

Dr15Jones commented:
The comparisons from the test all look very good. The only differences are the standard HGCal inconsistencies seen in IB and PR tests.

jfernan2 commented:
type e-gamma,jetmet
Waiting ofr JetMET and EGM conveners to comment

jfernan2 commented:
assign jetmet-pog

cmsbuild commented:
New categories assigned: jetmet-pog

@alkaloge,@kirschen,@miquork you have been requested to review this Pull request/Issue and eventually sign? Thanks

jfernan2 commented:
assign egamma-pog

cmsbuild commented:
New categories assigned: egamma-pog

@afiqaize,@RSalvatico you have been requested to review this Pull request/Issue and eventually sign? Thanks

SanghyunKo commented:
@kodolova Hi, FYI, the default EGM cut-based IDs are produced using a VID module named `egmGsfElectronIDs`, and the names of each (Run 3) ID is listed in the [`cutBasedElectronID_Winter22_122X_V1_cff.py`](https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/ElectronIdentification/python/Identification/cutBasedElectronID_Winter22_122X_V1_cff.py). So the name of the `InputTag` should be like `egmGsfElectronIDs:cutBasedElectronID-RunIIIWinter22-V1-medium` (or `veto`, `loose`, `tight`, depending on your needs).

kodolova commented:
That's ok. I will try to change.

mandrenguyen commented:
> So what should be the fate of this PR?

With apologies to Chris who spent time fixing the code, it sounds to me that this module can now be deleted as Sam suggested.   Perhaps [jfernan2](https://github.com/jfernan2) can confirm for reco.

jfernan2 commented:
Yes, EGM certified this module is not being used, so I guess it may be deleted

Sam-Harper commented:
Yes apologies to Chris for fixing this but I think the best thing is to delete it so we dont have to maintain it anymore.  Sorry for the waste of your time Chris

makortel commented:
Who will remove the module?

jfernan2 commented:
I'd propose @SanghyunKo @cochando as EGM Reco Contacts
Thanks

makortel commented:
Any news when the module could be removed?

SanghyunKo commented:
@makortel Hi. I've added on our EGM to-do list but it's relatively on a lower priority (compared to preparing 2025 prompt data-taking conditions, etc). Do you have any timeline in your mind for cleaning up for this?

Dr15Jones commented:
The decision is someone should remove the class.
